ubg/src/search/AllFilters.js

```diff
@@ -36,94 +66,100 @@ export default function AllFilters({setPaginationCount, setGames}) {
+   */
   const handleClear = () => {
-    setMinAge(21);
-    setMinPlayer(1);
-    setMaxPlayer('8+');
-    setMinTime(5);
-    setMaxTime('240+');
     setInitialize(false);
+    setClear(true);
   };
+  /**
+   * Load the games data according to filter
+   */
   useEffect(() => {
-    // using a hack to make useEffect act as onLoad()
-    if (initialize === false) {
+    // using a hack to load data and reset filters when needed
+    if (clear === true) {
+      setMinAge(21);
+      setMinPlayer(1);
+      setMaxPlayer('8+');
+      setMinTime(5);
+      setMaxTime('240+');
+      setClear(false);
+    } else if (initialize === false) {
+      setInitialize(true);
       const newGames = [];
       let list = [];
+      let maxP = maxPlayer;
+      if (typeof maxP === 'string') {
+        maxP = Number.MAX_SAFE_INTEGER;
+      }
+      let maxT = maxTime;
+      if (typeof maxT === 'string') {
+        maxT = Number.MAX_SAFE_INTEGER;
+      }
+      if (minPlayer > maxP) {
+        setMaxPlayer(minPlayer);
+        maxP = minPlayer;
+      }
+      if (minTime > maxT) {
+        setMaxTime(minTime);
+        maxT = minTime;
+      }
+      const values = value.trim().split(/ +/);
+      let total = 0;
       ref.orderBy(sortBy, 'desc')
           .get()
           .then(function(querySnapshot) {
-            setInitialize(true);
             querySnapshot.forEach(function(doc) {
-              list.push(doc.data());
-              if (list.length === 12) {
-                newGames.push(list);
-                list = [];
+              if (doc.data()['minPlayer'] <= maxP &&
+              minPlayer <= doc.data()['maxPlayer'] &&
+              doc.data()['minPlaytime'] <= maxT &&
+              minTime <= doc.data()['maxPlaytime'] &&
+              doc.data()['minAge'] <= minAge) {
+                if (value === '') {
+                  list.push(doc.data());
+                  total += 1;
+                  if (list.length === 12) {
+                    newGames.push(list);
+                    list = [];
+                  }
+                } else if (checkMatch(values, doc.data()['Name'])) {
+                  list.push(doc.data());
+                  total += 1;
+                  if (list.length === 12) {
+                    newGames.push(list);
+                    list = [];
+                  }
+                }
               }
             });
             if (list.length !== 0) {
               newGames.push(list);
             }
             setGames(newGames);
+            setTotalGames(total);
             setPaginationCount(newGames.length);
           })
           .catch(function(error) {
             console.log('Error getting documents: ', error);
           });
     }
-  }, [ref, sortBy, setGames, setPaginationCount, initialize]);
+  }, [ref, sortBy, setGames, setPaginationCount, initialize,
+    value, setInitialize, totalGames, setTotalGames, maxPlayer,
+    maxTime, minAge, minPlayer, minTime, setClear, clear]);
```

Comment:
So is this running every time you change a filter?

Reply:
Every time I search and clear filter